zerver/lib/user_groups.py

```diff
@@ -404,8 +405,8 @@ def access_user_group_for_setting(
     current_setting_value: UserGroup | None = None,
 ) -> UserGroup:
     if isinstance(setting_user_group, int):
-        named_user_group = access_user_group_by_id(
-            setting_user_group, user_profile, for_read=True, for_setting=True
+        named_user_group = get_user_group_by_id_in_realm(
+            setting_user_group, user_profile.realm, for_read=False
```

Comment:
Why is this one switching to `for_read=False`?

Reply:
We previously had `for_setting` parameter which was passed as `True` to make sure that we lock the user group row, which is now acheived by passing `for_read` as `False`. I actually had to ask this before, do we need a separate `for_setting` parameter and pass `for_read=True`? We actually do not update the group fetched here and just use it for a setting, so we are just reading the object.

Reply:
Yes, it's important. I believe the intent is to lock the group to prevent it from being deleted here, as part of our controls to avoid having a deleted/deactivated group ever being used as a setting value.

Reply:
I was passing `for_read=False` so that the group is locked. But yeah I have just added `for_setting` parameter again now since that felt more clear.